modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/IndexOnStartupExecutor.java

```diff
@@ -58,7 +58,7 @@ public Indexer<?> addingService(
 		Indexer<?> indexer = _bundleContext.getService(serviceReference);
 
 		boolean indexerIndexOnStartup = GetterUtil.getBoolean(
-			serviceReference.getProperty(PropsKeys.INDEX_ON_STARTUP));
```

Comment:
@shuyangzhou Please, don't do this. It's not always enabled. There is a portal property that controls it.

Reply:
Why would you default to true? How does that consider the value of the portal property?

Reply:
Relying on `serviceReference.getProperty(PropsKeys.*`seems not correct here as only the ConfigurationModelIndexer defines explicitly that property (as false), otherwise it depends on the value of the portal property for each model.

Reply:
Replied you on the ticket. This is the right thing to do.

Reply:
Thanks for the explanation.

Reply:
I forgot about that block. :eyes: